Source/WebCore/platform/sql/SQLiteDatabase.cpp

```diff
@@ -673,16 +673,16 @@ bool SQLiteDatabase::turnOnIncrementalAutoVacuum()
         if (!statement)
             return false;
         autoVacuumMode = statement->columnInt(0);
-    }
 
-    // Check if we got an error while trying to get the value of the auto_vacuum flag.
-    // If we got a SQLITE_BUSY error, then there's probably another transaction in
-    // progress on this database. In this case, keep the current value of the
-    // auto_vacuum flag and try to set it to INCREMENTAL the next time we open this
-    // database. If the error is not SQLITE_BUSY, then we probably ran into a more
-    // serious problem and should return false (to log an error message).
-    if (lastError() != SQLITE_ROW)
-        return false;
+        // Check if we got an error while trying to get the value of the auto_vacuum flag.
+        // If we got a SQLITE_BUSY error, then there's probably another transaction in
+        // progress on this database. In this case, keep the current value of the
+        // auto_vacuum flag and try to set it to INCREMENTAL the next time we open this
+        // database. If the error is not SQLITE_BUSY, then we probably ran into a more
+        // serious problem and should return false (to log an error message).
+        if (lastError() != SQLITE_ROW)
```

Comment:
Can you please add a comment explaining that the call to lastError() needs to occur while the transaction is still active/alive. Hopefully, it will discourage somebody from breaking this again in the future.